netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,215 @@
+
+        [Output]
+        public ITaskItem[] ConflictingReferences { get; set; }
+        [Output]
+
+        public ITaskItem[] ConflictingCopyLocalItems { get; set; }
+
+
+        public override bool Execute()
+        {
+            var conflictingReferences = new List<ITaskItem>();
+            HandleConflicts(ResolvedReferences, conflictingReferences);
+            ConflictingReferences = conflictingReferences.ToArray();
+
+            var conflictingCopyLocalItems = new List<ITaskItem>();
+            HandleConflicts(ResolvedCopyLocalItems, conflictingCopyLocalItems);
+            ConflictingCopyLocalItems = conflictingCopyLocalItems.ToArray();
+
+            return !Log.HasLoggedErrors;
+        }
+
+        void HandleConflicts(ITaskItem[] items, ICollection<ITaskItem> conflicts)
+        {
+            var targetPathToItem = new Dictionary<string, ITaskItem>(StringComparer.OrdinalIgnoreCase);
+            foreach (var item in items)
+            {
+                var targetPath = GetTargetPath(item);
+
+                ITaskItem existingItem;
+
+                if (targetPathToItem.TryGetValue(targetPath, out existingItem))
+                {
+                    // a conflict was found, determine the winner.
+                    var winner = HandleConflict(existingItem, item);
+
+                    if (winner != existingItem)
+                    {
+                        // replace existing item and add it to conflict list.
+                        targetPathToItem[targetPath] = item;
+                        conflicts.Add(existingItem);
+                    }
+                    else
+                    {
+                        // no need to replace item, just add new item to conflict list.
+                        conflicts.Add(item);
+                    }
+                }
+                else
+                {
+                    targetPathToItem[targetPath] = item;
+                }
+            }
+        }
+
+
+        private static string GetSourcePath(ITaskItem item)
+        {
+            var sourcePath = item.GetMetadata("HintPath");
+
+            if (String.IsNullOrWhiteSpace(sourcePath))
+            {
+                sourcePath = item.ItemSpec;
+            }
+
+            return sourcePath;
+        }
+
+        static readonly string[] s_targetPathMetadata = new[] { "TargetPath", "Path" };
+        private static string GetTargetPath(ITaskItem item)
+        {
+            foreach (var metadata in s_targetPathMetadata)
+            {
+                var targetPath = item.GetMetadata("TargetPath");
+
+                if (!String.IsNullOrWhiteSpace(targetPath))
+                {
+                    return targetPath.Replace('\\', '/');
+                }
+            }
+
+            var sourcePath = GetSourcePath(item);
+
+            return Path.GetFileName(sourcePath);
+        }
+
+
+        private ITaskItem HandleConflict(ITaskItem item1, ITaskItem item2)
```

Comment:
That's fair, and what I was getting at with my previous question.  I'll update it.